python/tank_vendor/shotgun_authentication/user_impl.py

```diff
@@ -276,6 +300,19 @@ def create_sg_connection(self):
             http_proxy=self._http_proxy,
         )
 
+    def are_credentials_valid(self):
+        """
+        Checks if the credentials for the user are valid.
+
+        :returns: True if the credentials are valid, False otherwise.
+        """
+        sg = self.create_sg_connection()
```

Comment:
Nobody is calling this yet for script users. I'm only providing it because it is implemented on the User public class. I think it's nice to have it the way it is anyway. This allows you to make sure your credentials in a clean way, regardless of the type of user you have.